File: Experiment/ServerBase/sensorsimulator/src/main/java/com/swot/sensorsimulator/example/SensorReading.java
package com.swot.sensorsimulator.example;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class SensorReading {
    @JsonProperty("id")
    private String nodeId;
    private double temperature;
    private double humidity;
    @JsonProperty("airQuality")
    private int aqi;
    @JsonProperty("noiseLevel")
    private double noise;
    private double latitude;
    private double longitude;
    private String zone;
    private long timestamp;

    // Constructor
    public SensorReading(String nodeId, double temperature, double humidity,
                         int aqi, double noise, double latitude, double longitude,
                         String zone, long timestamp) {
        this.nodeId = nodeId;
        this.temperature = temperature;
        this.humidity = humidity;
        this.aqi = aqi;
        this.noise = noise;
        this.latitude = latitude;
        this.longitude = longitude;
        this.zone = zone;
        this.timestamp = timestamp;
    }


}

File: Resources/python/sensors/src/models/sensor_reading.py
from dataclasses import dataclass
from datetime import datetime
from typing import Optional


@dataclass
class SensorReading:
    """Data class representing a sensor reading with all its attributes."""
    id: str
    temperature: float
    humidity: float
    airQuality: int
    noiseLevel: float
    latitude: float
    longitude: float
    zone: str
    timestamp: int

    @classmethod
    def create(cls,
               id: str,
               temperature: float,
               humidity: float,
               airQuality: int,
               noiseLevel: float,
               latitude: float,
               longitude: float,
               zone: str,
               timestamp: Optional[int] = None) -> 'SensorReading':
        """Factory method to create a SensorReading instance."""
        if timestamp is None:
            timestamp = int(datetime.now().timestamp() * 1000)

        return cls(
            id=id,
            temperature=temperature,
            humidity=humidity,
            airQuality=airQuality,
            noiseLevel=noiseLevel,
            latitude=latitude,
            longitude=longitude,
            zone=zone,
            timestamp=timestamp
        )
